src/blocks/image-comparison-item/components/Edit.js

```diff
@@ -22,12 +22,7 @@ const Edit = ({ attributes, setAttributes }) => {
   return (
     <>
       <Settings attributes={attributes} setAttributes={setAttributes} />
-      <Image
-        id={attributes?.id}
-        url={attributes?.url}
-        blockProps={blockProps}
-        setAttributes={setAttributes}
-      />
+      <Image blockProps={blockProps} attributes={attributes} setAttributes={setAttributes} />
```

Comment:
The pre-existing usage of the `Image` component may require a small tweak. Currently, it only returns a component with the `blockProps` applied to it's wrapper when an image is not selected. Also, the component is being re-used in the sidebar panel where you wouldn't normally expect a block to be rendered. My recommendation would be to lift the `<div {...blockProps}>` wrapper into the `Edit` component like so:

```suggestion
      <div {...blockProps}>
        <Image attributes={attributes} setAttributes={setAttributes} />
      </div>
```

This will allow the `Image` component to remain reusable and ensure only the instance used within the block list has the `blockProps` applied.

Reply:
Along the same lines, it may be beneficial to update the `Image` component to explicitly receive the props it requires, rather than passing the block attributes as an object

Reply:
changed in 05cd70f